clients/unshaded/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -61,7 +61,9 @@ public static BlockOutStream get(TachyonFile tachyonFile, WriteType opType, int
 
     if (tachyonFile.mTachyonFS.hasLocalWorker()
         && tachyonConf.getBoolean(Constants.USER_ENABLE_LOCAL_WRITE,
-            Constants.DEFAULT_USER_ENABLE_LOCAL_WRITE)) {
+            Constants.DEFAULT_USER_ENABLE_LOCAL_WRITE)
+            && tachyonConf.get(Constants.WORKER_BALANCER_STRATEGY_TYPE, "DEFAULT")
+                   .equalsIgnoreCase("DEFAULT")) {
```

Comment:
If `tachyonFile.mTachyonFS.hasLocalWorker()` returns `true`, it means the worker selection policy selects local worker(https://github.com/amplab/tachyon/blob/master/common/src/main/java/tachyon/worker/WorkerClient.java#L230), so I think there is no need to add other check here.